libs/deal-ticket/src/hooks/use-market-data.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo, useRef } from 'react';
+import type { MarketMarkPriceQuery } from './__generated__/MarketMarkPrice';
+import { useMarketMarkPriceQuery } from './__generated__/MarketMarkPrice';
+
+export const useMarketData = (marketId: string) => {
+  const memoRef = useRef<MarketMarkPriceQuery | null>(null);
+  const { data } = useMarketMarkPriceQuery({
+    pollInterval: 5000,
+    variables: { marketId },
+    skip: !marketId,
+  });
+  return useMemo(() => {
+    if (
+      data &&
+      data.market?.data?.markPrice !== memoRef.current?.market?.data?.markPrice
+    ) {
+      memoRef.current = data;
```

Comment:
non-blocking, question: wouldn't it be easier to hold in ref only `markPrice` for comparing?

Reply:
you're right. again, I haven't changed this hook, just moved it across. But, I think who wrote it, had the intention of returning more than mark price out of the market data.

Reply:
yeah, I remember we had in plans some revision of all these hooks, now seems to be a good opportunity, that's why I mentioned it here :wink: